scalyr_agent/configuration.py

```diff
@@ -29,6 +29,14 @@
 from scalyr_agent.json_lib import JsonConversionException, JsonMissingFieldException
 from scalyr_agent.util import JsonReadFileException
 
+import scalyr_agent.scalyr_logging as scalyr_logging
+
+# Set up the main logger.  We set it up initially to log to standard out,
+# but once we run fork off the daemon, we will use a rotating log file.
+log = scalyr_logging.getLogger('scalyr_agent')
+
+scalyr_logging.set_log_destination(use_stdout=True)
```

Comment:
You know, there's some dependency issues with using the scalyr logger from within this module.  Let's not import it here. Instead, have a reference to a logger passed into the method you need it below.. I think it is `parse`.  Default the argument to None (and don't log to it if is None).  However, then change the calls to `parse` in the agent main and other places to pass in the logger reference.